addons/binding/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/handler/KeContactHandler.java

```diff
@@ -473,51 +406,67 @@ public void run() {
         @Override
         public void run() {
             try {
+                long stamp = System.currentTimeMillis();
+                if (!InetAddress.getByName(((String) getConfig().get(IP_ADDRESS))).isReachable(PING_TIME_OUT)) {
+                    logger.debug("Ping timed out after '{}' milliseconds", System.currentTimeMillis() - stamp);
+                    logger.trace("Disconnecting the datagram channel '{}'", datagramChannel);
+                    try {
+                        datagramChannel.close();
+                    } catch (IOException e) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
```

Comment:
Any reason you continue here, because it will change the status again at 420, it looks like this try-catch block should be merged with the outer try-catch block.